keps/sig-multicluster/3335-statefulset-slice/README.md

```diff
@@ -0,0 +1,974 @@
+
+###### What are other known failure modes?
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+Downgrades are not gracefully supported, and are not backwards compatible. Cluster downgrades can cause a disruption to StatefulSet workloads if performed while `ordinals.start` is not `0`.
+
+## Alternatives
+
+<!--
+What other approaches did you consider, and why did you rule them out? These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+### Alternative API changes
+
+**ReverseOrderedReady**: A new PodManagementPolicy policy called
+`ReverseOrderedReady` could be added. This would allow a StatefulSet to be
+started and actuated from the highest ordinal (current default is from the
+lowest ordinal). For the cross-cluster migration use case, this would allow for
+a source StatefulSet to be scaled down and a target StatefulSet to be scaled in.
+The downside with this API is that pod management policy is not a mutable field.
+So if an orchestrator uses this behavior to scale in a StatefulSet, in a
+destination cluster, and then wants to revert the PodManagementPolicy back to
+default, the StatefulSet would need to be deleted, and re-created.
+
+### Alternatives without any API changes
+
+**Orphan Pods**: Users can orphan pods from a StatefulSet, migrate pods across a
+namespace or cluster, and create a new StatefulSet to manage pods upon
+migration. In the case of pod eviction or failure, pods will need to be manually
+restarted, requiring manual intervention and constant monitoring.
```

Comment:
`recreated` technically, would prefer that be updated (restarts isn't something we define as an action in kube, recreates of pods is)